src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriverCounterVisitor.java

```diff
@@ -0,0 +1,64 @@
+package edu.kis.powp.jobs2d.drivers.visitor;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.DriverMacro;
+import edu.kis.powp.jobs2d.drivers.PreciseLoggerDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+import edu.kis.powp.jobs2d.drivers.adapter.TrackedJob2dDriver;
+import edu.kis.powp.jobs2d.drivers.composite.DriverContainer;
+import edu.kis.powp.jobs2d.drivers.visitor.DriverVisitor;
+import edu.kis.powp.jobs2d.features.driverTransofrmation.TransformingDriver;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class DriverCounterVisitor implements DriverVisitor {
+
+    private int count = 0;
+
+    @Override
+    public void visitDriverContainer(DriverContainer driverContainer) {
+        count += 1;
+        for (Job2dDriver child : driverContainer.getChildren()) {
+            try {
+                Method acceptor = child.getClass().getMethod("accept", DriverVisitor.class);
+                acceptor.invoke(child, this);
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
OK, but such an approach is not stateless. The following visitor calls rely on the previous calls and the same visitor object cannot be reused. 
The better approach would be stateless (e.g. visitLineDriverAdapter should always set count = 1 because it is a single driver).

Reply:
Done